src/literal.h

```diff
@@ -96,7 +96,60 @@ class Literal {
     }
     return false;
   }
+  bool isZero() const {
+    if (type.isVector()) {
+      uint8_t zeros[16] = {0};
+      return memcmp(&v128, zeros, 16) == 0;
+    }
+    return isSingleZero();
+  }
+  bool isSingleZero() const {
+    switch (type.getBasic()) {
+      case Type::i32:
+        return i32 == 0;
+      case Type::i64:
+        return i64 == 0LL;
+      case Type::f32:
+        return bit_cast<float>(i32) == 0.0f;
+      case Type::f64:
+        return bit_cast<double>(i64) == 0.0;
+      default:
+        WASM_UNREACHABLE("unexpected type");
+    }
+  }
```

Comment:
Normally the `doX` and `doSingleX` pairs of functions operate on arbitrary types and non-tuple types, respectively, not on arbitrary number types and non-vector number types. Could we combine these back into just one `isZero` function or is splitting them up still useful for something that I'm missing? (And the same for `makeZero`/`makeSingleZero`)